csharp/ql/src/semmle/code/csharp/security/dataflow/ReDoS.qll

```diff
@@ -33,23 +34,9 @@ module ReDoS {
 
     override predicate isSource(DataFlow::Node source) { source instanceof Source }
 
-    override predicate isSink(DataFlow::Node sink) {
-      sink instanceof Sink
-      or
-      // Unfortunately, we cannot add `ExponentialRegexSink` as
-      // a sub class of `Sink`, as that results in bad aggregate
-      // recursion. Therefore, we overestimate the sinks here
-      // and make the restriction later by overriding
-      // `hasFlowPath()` below.
-      sink.asExpr() = any(RegexOperation ro).getInput()
-    }
+    override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
```

Comment:
I presume that the performance problem mentioned by the previous code has been addressed through the use of multiple copies of the dataflow library?

Reply:
Yes, now that the two configurations use different copies of the library it is safe to chain them together.